Automattic/jetpack
# Boost: Make Minify concat concat service independent of WordPress

> **Warning**
> Depends on another PR.
> Review and Merge after https://github.com/Automattic/jetpack/pull/30863.

## Proposed changes:

* Update the Minify concatenation service to work without WordPress.

### Other information:


## Jetpack product discussion



pc9hqz-1xD-p2#comment-1185

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






* Setup this version of the plugin on your website;
* Turn on CSS or JS concatenation;
* In order to test that the code works without WordPress, you need to add a snippet at the top of your website's root `index.php`:

```php
if ( isset( $_SERVER['REQUEST_URI'] ) && '/_static/' === substr( $_SERVER['REQUEST_URI'], 0, 9 ) ) {
	// Page Optimize setup
	define( 'PAGE_OPTIMIZE_CACHE_DIR', false ); // Disable cache
	// paths so we can run service.php w/o WP
	define( 'PAGE_OPTIMIZE_CONCAT_BASE_DIR', __DIR__ );
	define( 'PAGE_OPTIMIZE_ABSPATH', PAGE_OPTIMIZE_CONCAT_BASE_DIR );

	require_once __DIR__ . '/wp-content/plugins/boost/serve-minified-content.php';
	exit;
}
```

* Update `require_once __DIR__ . '/wp-content/plugins/boost/serve-minified-content.php';` accordingly;
* Now open the home page and check to see that everything works as expected (styles and JS should be okay).
